pydy/viz/scene.py

```diff
@@ -186,109 +153,104 @@ def generate_visualization_json(self, dynamic_variables,
-                                                           constant_variables,
-                                                           dynamic_values,
-                                                           constant_values)
-        self._scene_data_dict = self.generate_scene_dict(constant_map=constant_map)
-        self._scene_data_dict["simulationData"] = self.simulation_json_file
+        constant_map_for_json = dict(zip(constant_variables_str,
+                                         constant_values))
 
-        self._scene_data_dict["timeDelta"] = 1/fps
-        self._scene_data_dict["timeSteps"] = len(dynamic_values)
+        self._simulation_data_dict = \
+            self.generate_simulation_dict(dynamic_variables,
+                                          constant_variables,
+                                          dynamic_values,
+                                          constant_values)
 
+        self._scene_data_dict = \
+            self.generate_scene_dict(constant_map=constant_map)
 
+        self._scene_data_dict["simulationData"] = self.simulation_json_file
+        self._scene_data_dict["timeDelta"] = 1 / fps
+        self._scene_data_dict["timeSteps"] = dynamic_values.shape[0]
         self._scene_data_dict["constant_map"] = constant_map_for_json
-        scene_data_outfile = open(self.scene_json_file, 'w')
-        scene_data_outfile.write(json.dumps(self._scene_data_dict, indent=4,
-                                 separators=(',', ': ')))
-        scene_data_outfile.close()
 
-        simulation_data_outfile = open(self.simulation_json_file, 'w')
-        simulation_data_outfile.write(json.dumps(self._simulation_data_dict, indent=4,
-                                 separators=(',', ': ')))
-        simulation_data_outfile.close()
+        with open(self.scene_json_file, 'w') as scene_data_outfile:
+            scene_data_outfile.write(json.dumps(self._scene_data_dict,
+                                                indent=4,
+                                                separators=(',', ': ')))
+
+        with open(self.simulation_json_file, 'w') as simulation_data_outfile:
+            simulation_data_outfile.write(json.dumps(
+                self._simulation_data_dict, indent=4,
+                separators=(',', ': ')))
 
     def generate_scene_dict(self, constant_map={}):
-        """
-        This method is used to create the dictionary compatible with
-        PyDy visualizer. This JSON file contains all the relevant information
-        required by PyDy visualizer to draw the scene on the canvas.
+        """Generates a dictionary containing all of the information needed
+        to build the scene.
 
+        Parameters
+        ==========
+        constant_map : dictionary
+            A map of symbolic constants to numerical values. This is
+            typically used if there are symbolics in the dimensions of the
+            shapes.
```

Comment:
"dimensions of the shapes" can be confusing. At first I thought you meant like the dimensionality of a vector. Maybe "parameters of the shapes"? idk.